### ISSUE TITLE
FNX2-15457 ⁃ Releng Firefox Beta 79.1.0

### ISSUE THREAD

**Issue by liuche** [mozilla-mobile/fenix#12761]
This build is currently using 79.0.0 Beta, but 80 will go to Beta later this week. If we want to update, we can push a new version.

## Beta Builds
* [x] ~~Monday, July 20 Firefox Beta build w/ AC 50.0.0~~ Tuesday, July 21 Firefox Beta build w/ AC 51.0.0

### During Beta Product Integrity (Beta Release until PI green signoff)
* [ ] If bugs are considered release blocker then find someone to fix them on master and the milestone branch (cherry-pick / uplift)
* [ ] If needed tag a new RC version (e.g. v1.0-RC2) and follow the submission checklist again.

### During Production Release Rollout [Tuesday, Wednesday following Monday Release Day]
* [ ] Check Sentry for new crashes. File issues and triage.
* [ ] Ask Relman in the bug if they see potential blockers on Google Play, and if not, request that they bump the release each day (25% Tu, 100% Wed)

```
Major releases often need to be synchronized with other marketing activities (e.g. blog postings).
```

## Room for improvement
* [ ] Automate assigning milestones to closed issues (based on date, etc) #6199
* [ ] Automate assignig `eng:qa:needed` to issues #6199
* [ ] Automate verification that the commit hash matches the most recent RC
* [ ] Builds generated as part of `signing-production` task look like `public/build/arm64-v8a/geckoBeta/target.apk`.  This means that the dev must download, then rename them by hand.  Could RM update these to generate `public/build/arm64-v8a/geckoBeta/firefox-preview-v3.0.0-rc.1-arm64-v8a.apk`, or similar?

**Comment by liuche** [comment]
[#12761, #12687, #12683, #12517, #12677, #12681, #12682, #7966, #12667, #12532, #12680, #9949, #12630, #12652, #12509, #12453, #12364, #12649, #12640, #12531, #12654, #10173, #11208, #12457, #12612, #12507, #12581, #12565, #12579, #12600, #12601, #12201, #12543, #12603, #12118, #12248, #11704, #11965, #12493, #12105, #12423, #12544, #12560, #12520, #12567, #10806, #12503, #7094, #12262, #12171, #12555, #12437, #12524, #12107, #11886, #12258, #11099, #12189, #7532, #10466, #12106, #10911, #11875, #12407, #11655, #12136, #12464, #11324, #12222, #44, #12339, #12433, #11303, #9832, #11014, #12010, #12429, #12311, #12336, #12394, #12404, #11563, #11431, #12353, #12348, #11522, #12370, #11873, #12312, #12345, #12358, #12126, #11493, #11763, #11889, #12261, #12078, #9775, #12263, #12007, #12016, #12086, #12324, #12125, #12316, #11764, #12309, #12314, #10401, #12254, #12018, #12252, #12250, #8714, #12251, #12224, #12233, #11320, #12207, #12209, #11183, #12157, #12175, #12177, #12169, #12095, #12096, #11996, #12080, #12173, #12172, #11642, #12142, #9946, #10066, #11872, #12127, #12081, #12094, #11991, #12055, #12013, #11814]

**Comment by liuche** [comment]
Tried to pin to AC 50.0.0 from last week, but it seems like 51.0.0 has added a ton of breaking changes so we'll wait for the AC 51.0.0 release on Tuesday cc @pocmo - since this was cut in the morning, AC 51.0.0 should apply cleanly.

In the future, we'll be ending our sprints on Wednesday EOD, so we should be able to pick the freshest AC at the end of every sprint 🥬

**Comment by pocmo** [comment]
AC 51.0.0 is now released and available.

But there's something odd about the versioning here. This is a new branch that will be cut from `master`, right? Using `79.0.1` for that seems wrong, I would expect this to be a dot release on top of `79.0.0` and not a new version from `master`.

It also gets confusing with the release version: We are going to release 79.0.0 to the fennec release audience soon. If we need to follow-up with a dot release to fix something then this would be 79.0.1 - which then would be something completely different than 79.0.1-Beta that we build here. This is very confusing.

This also is somehow misleading with semantic versioning: This is not a [patch version with "backwards compatible bug fixes."](https://semver.org/).

We can't really change the major version while it is syncrhonized with Gecko(View). But maybe we should bump the minor version here instead and turn this into `79.1.0`? This also makes it possible to continue with "patch" dot releases from this version.

**Comment by liuche** [comment]
Rolled out to 100% last week.
